geopyspark/geotrellis/catalog.py

```diff
@@ -475,19 +286,39 @@ class AttributeStore(object):
         hist = store.layer("us-nlcd2011-30m-epsg3857", zoom=7).read("histogram")
         hist = gps.Histogram.from_dict(hist)
     """
-
     def __init__(self, uri):
         self.uri = uri
-        self.underlying = scala_companion("geotrellis.spark.io.AttributeStore").apply(uri)
+        pysc = get_spark_context()
+        try:
+            self.wrapper = pysc._gateway.jvm.geopyspark.geotrellis.io.AttributeStoreWrapper(uri)
+        except Py4JJavaError as err:
+            raise ValueError(err.java_exception.getMessage())
+
+
+    @classmethod
+    def build(cls, uri):
```

Comment:
Sorry, I was wrong about the `__slots__`. However, I'm still not sure this method is needed. Why can't the logic in `build` be moved to `__init__`?